bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/dto/clip2/Resource.java

```diff
@@ -0,0 +1,615 @@
+        return temperature != null ? temperature.getTemperatureValidState() : UnDefType.UNDEF;
+    }
+
+    public @Nullable Effects getTimedEffects() {
+        return timedEffects;
+    }
+
+    public ResourceType getType() {
+        return ResourceType.of(type);
+    }
+
+    public State getZigBeeState() {
+        ZigBeeStatus zigBeeStatus = getZigBeeStatus();
+        return zigBeeStatus != null ? new StringType(zigBeeStatus.toString()) : UnDefType.UNDEF;
+    }
+
+    public @Nullable ZigBeeStatus getZigBeeStatus() {
+        String status = this.status;
+        return status != null ? ZigBeeStatus.of(status) : null;
+    }
+
+    public boolean hasFullState() {
+        return !hasSparseData;
+    }
+
+    /**
+     * Mark that the resource has sparse data.
+     *
+     * @return this instance.
+     */
+    public Resource markAsSparse() {
+        hasSparseData = true;
+        return this;
+    }
+
+    /**
+     * Set the brightness percent.
+     *
+     * @param command either a PercentType with the new value, an OnOffType to set it at 0 / 100 percent, or an
+     *            IncreaseDecreaseType to increment the percentage value by a fixed amount.
+     * @return this resource instance.
+     */
+    public Resource setBrightness(Command command) {
+        State state = getPercentType(command, getBrightnessState());
+        if (state instanceof PercentType) {
+            Dimming dimming = this.dimming;
+            dimming = dimming != null ? dimming : new Dimming();
+            dimming.setBrightness(((PercentType) state).intValue());
+            if (PercentType.ZERO.equals(state)) {
+                setSwitch(OnOffType.OFF);
+            }
+            this.dimming = dimming;
+        }
+        return this;
+    }
+
+    /**
+     * Set the color from an HSBType. Put its Hue & Saturation parts in the 'ColorXy' JSON element, and put its
+     * Brightness part in the 'Dimming' JSON element.
+     *
+     * @param command an HSBType with the new color value
+     * @return this resource instance.
+     */
+    public Resource setColor(Command command) {
+        if (command instanceof HSBType) {
+            HSBType hsb = (HSBType) command;
+            ColorXy col = color;
+            Dimming dim = dimming;
+            color = (col != null ? col : new ColorXy()).setXY(getColorXY(hsb));
+            dimming = (dim != null ? dim : new Dimming()).setBrightness(hsb.getBrightness().intValue());
+        }
+        return this;
+    }
+
+    /**
+     * Set the colour temperature in Kelvin.
+     *
+     * @param command should be a QuantityType(Temperature> (but it can also handle DecimalType).
+     * @return this resource instance.
+     */
+    public Resource setColorTemperatureKelvin(Command command) {
+        Integer kelvin = null;
+        if (command instanceof QuantityType<?>) {
+            QuantityType<?> temperature = ((QuantityType<?>) command).toUnit(Units.KELVIN);
```

Comment:
You should use toInvertibleUnit here, otherwise it won't be able to convert from mired/mirek to Kelvin

Reply:
Done.